File: 2023_us_open/pareidolia.py
s = input()
ans = 0
dp = [0 for i in range(4)]
sol = 0
for i in s:
    if (i == 'b'):
        dp[0] += 1
    elif (i == 's'):
        dp[2] += 1
    elif (i=='i'):
        dp[3] += 1
    elif (i == 'e'):
        dp[1] += 1
        tof = True
        for j in range(4):
            cur = dp[j]
            if (j == 0):
                if (cur == 0):
                    tof = False
                    break
            elif (j == 1):
                if (cur < 2):
                    tof = False
                    break
            elif (j == 2):
                if (cur < 2):
                    tof = False
                    break
            else:
                if (cur == 0):
                    tof = False
                    break
        if (tof):
            sol += 1
            for j in range(4):
                if j == 0 or j == 3:
                    dp[j] -= 1
                else:
                    dp[j] -= 2
    print(dp)
    ans += sol
print(ans)

File: 2023_us_open/pareidolia.cpp
#include <bits/stdc++.h>

using namespace std;

int main()  {
    string a;
    cin >> a;
    int s = a.length();
    vector<long long> dp(6);
    long long ans = 0;
    long long sol = 0;
    for (int i = 0; i < s; i++) {
        dp[0] += 1;
        if (a[i] == 'b')    {
            dp[1] += dp[0];
            dp[0] = 0;
        } else if (a[i] == 'e') {
            dp[2] += dp[1];
            ans += dp[5];
            dp[0] += dp[5];
            dp[1] = 0;
            dp[5] = 0;
        } else if (a[i] == 's') {
            dp[4] += dp[3];
            dp[3] = dp[2];
            
            dp[2] = 0;
        } else if (a[i] == 'i') {
            dp[5] += dp[4];
            dp[4] = 0;
        }
        //cout << dp[0] << ' ' << dp[1] << ' ' << dp[2] << ' ' << dp[3] << ' ' << dp[4] << ' ' << dp[5] << endl;
        //cout << "ans" << ' ' << ans << ' '<< sol << endl;
        sol += ans; 
    }
    cout << sol << endl;
}